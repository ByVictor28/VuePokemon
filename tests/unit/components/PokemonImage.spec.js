import { shallowMount } from "@vue/test-utils"
import PokemonPicture from "@/components/PokemonImage"

describe("PokemonImage componebt",()=>{
  let wrapper

  beforeEach(()=>{
    wrapper = shallowMount(PokemonPicture,{
      props:{
        pokemonId:100,
        showPokemon:false
      }
    })
  })
  
  test('snapshot match', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('show hidden image and pokemon 100', () => {
    const wrapper = shallowMount(PokemonPicture,{
      props:{
        pokemonId:100,
        showPokemon:false
      }
    })

    const [img1,img2] = wrapper.findAll("img")

    expect(img1.exists()).toBeTruthy()
    expect(img2).toBe(undefined)
    console.log(img1.classes())
    expect(img1.classes("brightness-0")).toBeTruthy()

    expect(img1.attributes("src")).toBe(`https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/${100}.svg`)
  })

  test('show pokemon image, pokemon 100 when showPokemon=true', () => {
    const wrapper = shallowMount(PokemonPicture,{
      props:{
        pokemonId:100,
        showPokemon:true
      }
    })

    const [img1,img2] = wrapper.findAll("img")

    expect(img1.exists()).toBeTruthy()
    expect(img2.exists()).toBeTruthy()
    expect(img2.classes("fade-in")).toBeTruthy()
    
  })
  
})