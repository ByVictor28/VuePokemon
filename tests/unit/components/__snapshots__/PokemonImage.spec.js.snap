// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonImage componebt snapshot match 1`] = `
<div class="">
  <div class="select-none w-full h-52 flex flex-col justify-center items-center "><img class="filter brightness-0 w-52 h-52 absolute select-none" src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg" alt="Pokemon Shadow">
    <!--v-if-->
  </div>
</div>
`;
